pkg/objstore/metric_bucket.go

```diff
@@ -0,0 +1,203 @@
+package objstore
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// BucketWithMetrics takes a bucket and registers metrics with the given registry for
```

Comment:
What is this, how this is related to the adding retries and backoff?

Reply:
hm?

Reply:
I just moved it from objstore.go https://github.com/improbable-eng/thanos/blob/master/pkg/objstore/objstore.go#L172, as it's a separate entity